src/pojo-interfaces.test.ts

```diff
@@ -0,0 +1,332 @@
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { describe, it, expect } from "@jest/globals";
+import { dataset } from "@rdfjs/dataset";
+import * as fc from "fast-check";
+import { DataFactory } from "n3";
+import {
+  serializeBoolean,
+  serializeDatetime,
+  serializeDecimal,
+  serializeInteger,
+  xmlSchemaTypes,
+} from "./datatypes";
+import { fromRdfJsDataset, toRdfJsDataset } from "./pojo-interfaces";
+
+describe("fromRdfJsDataset", () => {
+  const fcNamedNode = fc.webUrl().map((url) => DataFactory.namedNode(url));
+  const fcString = fc.string().map((value) => DataFactory.literal(value));
+  const fcInteger = fc
+    .integer()
+    .map((value) =>
+      DataFactory.literal(
+        serializeInteger(value),
+        DataFactory.namedNode(xmlSchemaTypes.integer)
+      )
+    );
+  const fcDecimal = fc
+    .float()
+    .map((value) =>
+      DataFactory.literal(
+        serializeDecimal(value),
+        DataFactory.namedNode(xmlSchemaTypes.decimal)
+      )
+    );
+  const fcDatetime = fc
+    .date()
+    .map((value) =>
+      DataFactory.literal(
+        serializeDatetime(value),
+        DataFactory.namedNode(xmlSchemaTypes.dateTime)
+      )
+    );
+  const fcBoolean = fc
+    .boolean()
+    .map((value) =>
+      DataFactory.literal(
+        serializeBoolean(value),
+        DataFactory.namedNode(xmlSchemaTypes.boolean)
+      )
+    );
+  const fcLangString = fc
+    .tuple(
+      fc.string(),
+      fc.oneof(fc.constant("nl-NL"), fc.constant("en-GB"), fc.constant("fr"))
+    )
+    .map(([value, lang]) => DataFactory.literal(value, lang));
+  const fcArbitraryLiteral = fc
+    .tuple(fc.string(), fc.webUrl())
+    .map(([value, dataType]) =>
+      DataFactory.literal(value, DataFactory.namedNode(dataType))
+    );
+  const fcLiteral = fc.oneof(
+    fcString,
+    fcInteger,
+    fcDecimal,
+    fcDatetime,
+    fcBoolean,
+    fcLangString,
+    fcArbitraryLiteral
+  );
+  // To check: not sure whether this generates new blank nodes every time.
+  const fcBlankNode = fc.constant(null).map(() => DataFactory.blankNode());
+  const fcDefaultGraph = fc.constant(DataFactory.defaultGraph());
+  const fcGraph = fc.oneof(fcDefaultGraph, fcNamedNode);
+  const fcQuadSubject = fc.oneof(fcNamedNode, fcBlankNode);
+  const fcQuadPredicate = fcNamedNode;
+  const fcQuadObject = fc.oneof(fcNamedNode, fcLiteral, fcBlankNode);
+  const fcQuad = fc
+    .tuple(fcQuadSubject, fcQuadPredicate, fcQuadObject, fcGraph)
+    .map(([subject, predicate, object, graph]) =>
+      DataFactory.quad(subject, predicate, object, graph)
+    );
+  const fcTerm = fc.oneof(fcNamedNode, fcLiteral, fcBlankNode, fcQuad);
```

Comment:
Yeah I figured this might be useful for that, but then I realised that comparing different instances of RDF/JS Datasets was a chore and did not implement that.